### DISCUSSION TITLE
I need to use prisma in golang

### DISCUSSION THREAD

**Discussion by AtegyekaRodgers** [prisma/prisma#6802] [Q&A]
I have tried to follow the instructions on a github repo: 
https://github.com/prisma/prisma-client-go
but the commands they told me to run do not succed.
When I run the command 
`$ go run github.com/prisma/prisma-client-go db push --preview-feature
`
I get the follwing error:
```
┌─[✗]─[rodgers@parrot]─[~/Projects/myproject]
└──╼ $go run github.com/prisma/prisma-client-go generate
info: 2021/04/26 19:20:22 prisma cli doesn't exist, fetching... (this might take a few minutes)
panic: could not fetch binaries: 
could not download engines: 
could not download https://prisma-photongo.s3-eu-west-1.amazonaws.com/prisma-cli-2.18.0-linux.gz to 
/home/rodgers/.cache/prisma/binaries/cli/2.18.0/prisma-cli-linux: 
could not get https://prisma-photongo.s3-eu-west-1.amazonaws.com/prisma-cli-2.18.0-linux.gz: 
Get "https://prisma-photongo.s3-eu-west-1.amazonaws.com/prisma-cli-2.18.0-linux.gz": 
read tcp 10.127.58.8:60518->52.218.57.96:443: read: connection reset by peer

goroutine 1 [running]:
main.main()
	/home/rodgers/Projects/myproject/build/pkg/mod/github.com/prisma/prisma-client-go@v0.7.0/main.go:29 
+0x493
exit status 2

```

**Comment by janpio** [comment]
Can you download these files like https://prisma-photongo.s3-eu-west-1.amazonaws.com/prisma-cli-2.18.0-linux.gz manually? This works perfectly fine for me, so we need to exclude that your connection is weird in some way.

**Reply by AtegyekaRodgers** [reply]
I have now downloaded manually, but am not sure of what next after that ?
How can I make it work as it was supposed to with the first approach which failed ?

**Reply by janpio** [reply]
That was not the goal. We just wanted to understand if you can in general access these files. Now we need to figure out, why the Go Client can not do that. Can you run `curl https://prisma-photongo.s3-eu-west-1.amazonaws.com/prisma-cli-2.18.0-linux.gz` in the same terminal where you were trying to run the `go run` command please?

(cc @steebchen)

**Reply by steebchen** [reply]
Also, after downloading the file manually, did you simply try running the go client command again? It looks like a temporary network issue to me, especially if downloading the file manually works. Please let us know if you still run into the same issue.

**Reply by AtegyekaRodgers** [reply]
```
┌─[rodgers@parrot]─[~/Projects/myproject]
└──╼ $curl https://prisma-photongo.s3-eu-west-1.amazonaws.com/prisma-cli-2.18.0-linux.gz
curl: (35) OpenSSL SSL_connect: Connection reset by peer in connection to prisma-photongo.s3-eu-west-1.amazonaws.com:443 
┌─[✗]─[rodgers@parrot]─[~/Projects/myproject]
└──╼ $

**Reply by janpio** [reply]
That indicates that the terminal you are running in can not connect to these servers for some reasons. Maybe a proxy or enterprise security thing running there? This should definitely work.

**Reply by AtegyekaRodgers** [reply]
Is it because I don't have SSL configured on my laptop ?
